docs/upgrade-workflow.md

```diff
@@ -0,0 +1,423 @@
+# Introduction
+
+The purpose of this document is documenting observations made during the upgrade
+process of various kubernetes components using the metal3-dev-env environment.
+The following list contains the experiments performed and their outcomes. The
+list could grow or shrink overtime as we learn more about the process.
+
+ If you have use cases to be covered, observations of your own or any input,
+ please do create a pull request with content in the relevant sections or a
+ section of your own.
+
+As to help others to perform the same experiments, please do divide your work
+into sections as shown in the ```Outcomes and observations```. Briefly, it should
+include the versions of different kubernetes components before and after the
+upgrade process. Also relevant is documenting environmental issues noticed during
+the experiments, such as delays of some operations.
+
+# Experiments
+
+We cover different cases of upgrade. Broadly speaking, we focus on the following
+cases and add sub cases as needed.
+
+- Upgrade of kubernetes version
+- Upgrade of kubernetes binaries, kubeadm, kubelet and kubectl
+- Upgrade of boot disk image
+- Upgrade of a single master
+  - Version 1 -> Version 2
+  - Version 1 -> Version 2 -> Version 3
+- Upgrade of multiple masters
+  - Upgrade of masters with one extra Server
+  - Upgrade of masters with no extra server
```

Comment:
possible when https://github.com/kubernetes-sigs/cluster-api/pull/2688 is merged